src/check/constrain/generate/control_flow.rs

```diff
@@ -20,19 +19,22 @@ pub fn gen_flow(
 ) -> Constrained {
     match &ast.node {
         Node::Handle { expr_or_stmt, cases } => {
-            let raises: HashSet<TrueName> = cases.iter().flat_map(|c| match &c.node {
+            let (raises, errs): (Vec<Result<_, _>>, Vec<Result<_, _>>) = cases.iter().map(|c| match &c.node {
                 Node::Case { cond, .. } => {
                     match &cond.node {
-                        Node::ExpressionType { ty: Some(ty), .. } => if let Node::Type { .. } = &ty.node {
-                            if let Ok(name) = TrueName::try_from(ty) {
-                                Some(name)
-                            } else { None }
-                        } else { None }
-                        _ => None
+                        Node::ExpressionType { ty: Some(ty), .. } => TrueName::try_from(ty)
+                            .map_err(|errs| errs.first().expect("At least one").clone()),
+                        other => {
+                            let msg = format!("Expected type identifier, was {other}");
```

Comment:
Unreachable arm because this is already covered in parse stage.
Perhaps we in future should have a special type of node which can be converted to a `Name` without any error.